internal/server.go

```diff
@@ -11,50 +11,142 @@ import (
+      margin: auto;
+      box-sizing: border-box;
+      background-color: #eee;
+    }
+
+    #grid {
+      display: grid;
+      grid-gap: 1rem;
+      grid-template-columns: 1fr 1fr;
+      grid-template-rows: repeat(3, 2rem);
+      margin-bottom: 1rem;
+    }
+
+    #grid > * {
+      margin: auto 0;
+    }
+
+    input, button{
+      width: 100%;
+    }
+  </style>
+</head>
+
+<body>
+  <h1># gosh! Go Share</h1>
+  <p>
+    Upload your files to this server and share them with your friends or, if
+    non-existent, shady people from the Internet.
+  </p>
+  <p>
+    Your file will expire after
+    {{.Expires}} or earlier, if explicitly specified. Optionally, the file can
+    be deleted directly after the first retrieval. In addition, the maximum file
+    size is {{.Size}}. 
+  </p>
+  <p>
+    This is no place to share questionable or illegal data.
+    Please use another service or stop it completely. Get some help.
+  </p>
+  <p>
+    The gosh
+    software can be obtained from
+    <a href="https://github.com/oxzi/gosh">https://github.com/oxzi/gosh</a>
+  </p>
+
+  <h2>## Posting</h2>
+
+  <h3>### curl</h3>
+
+  HTTP POST your file:
+
+  <pre>$ curl -F 'file=@foo.png' {{.Proto}}://{{.Hostname}}/</pre>
+
+  Burn after reading:
+
+  <pre>$ curl -F 'file=@foo.png' -F 'burn=1' {{.Proto}}://{{.Hostname}}/</pre>
+
+  Set a custom expiry date, e.g., one minute:
+
+  <pre>
+$ curl -F 'file=@foo.png' -F 'time=1m' {{.Proto}}://{{.Hostname}}/</pre>
+
+  Or all together:
+
+  <pre>
+$ curl -F 'file=@foo.png' -F 'time=1m' -F 'burn=1' {{.Proto}}://{{.Hostname}}/</pre>
+
+  <h3>### form</h3>
+  
+  <form action="{{.Proto}}://{{.Hostname}}/" method="POST" enctype="multipart/form-data">
+    <div id="grid">
+      <label for="file">Your file:</label>
+      <input type="file" name="file" />
+      <label for="burn">Burn after reading:</label>
+      <input type="text" name="burn" pattern="1?" title="Use '1' to burn, leave empty to retain your file" />
+      <label for="time">Optionally, set a custom expiry date:</label>
+      <input
+        type="text"
+        name="time"
+        pattern="[-+]?([0-9]+(\.[0-9]*)?(ns|us|ms|s|m|h)+)+"
+        title="A duration string is a possibly signed sequence of decimal numbers,
+               each with optional fraction and a unit suffix such as '300ms',
+               '-1.5h' or '2h45m'. Valid time units are 'ns', 'us' (or 'µs') 'ms',
+               's', 'm', 'h'"
+      />
+    </div>
+    <button>Upload</button>
+  </form>
+  <h2>## Privacy</h2>
+
+  This software stores the IP address for each upload. This information is
+  stored as long as the file is available. In addition, the IP address of the
+  user might be loged in case of an error. A normal download is logged without
+  user information.
+
+  <h2>## Abuse</h2>
+
+  If, for whatever reason, you would like to have a file removed prematurely,
+  please write an e-mail to <{{.EMail}}>. Please allow me a certain amount of
```

Comment:
The `.EMail` field will result in an invalid HTML tag, e.g., `foo@example.org` results in `…please write an e-mail to <foo@example.org>.`. You might want to change this either to a `mailto` link or `<pre>` the e-mail address?